apps/extension/src/ui/domains/ViewOnExplorer/ExplorerNetworkPicker.tsx

```diff
@@ -0,0 +1,156 @@
+import { ScrollContainer } from "@talisman/components/ScrollContainer"
+import { SearchInput } from "@talisman/components/SearchInput"
+import { ExternalLinkIcon, XIcon } from "@talismn/icons"
+import { isEthereumAddress } from "@talismn/util"
+import { useAccountByAddress } from "@ui/hooks/useAccountByAddress"
+import useBalancesByAddress from "@ui/hooks/useBalancesByAddress"
+import { useBalancesFiatTotalPerNetwork } from "@ui/hooks/useBalancesFiatTotalPerNetwork"
+import useChains from "@ui/hooks/useChains"
+import { useEvmNetworks } from "@ui/hooks/useEvmNetworks"
+import { useSetting } from "@ui/hooks/useSettings"
+import { isAccountCompatibleWithChain } from "@ui/util/isAccountCompatibleWithChain"
+import { isAddressCompatibleWithChain } from "@ui/util/isAddressCompatibleWithChain"
+import { FC, useCallback, useMemo, useState } from "react"
+import { useTranslation } from "react-i18next"
+import { IconButton } from "talisman-ui"
+import urlJoin from "url-join"
+
+import { ChainLogo } from "../Asset/ChainLogo"
+
+type NetworkWithExplorer = {
+  id: string
+  type: "evm" | "substrate"
+  name: string
+  explorerUrl: string
+}
+
+const useExplorerNetworks = (address: string, search: string): NetworkWithExplorer[] => {
+  const account = useAccountByAddress(address)
+  const [includeTestnets] = useSetting("useTestnets")
+  const { chains } = useChains({ activeOnly: true, includeTestnets })
+  const { evmNetworks } = useEvmNetworks({ activeOnly: true, includeTestnets })
+  const balances = useBalancesByAddress(address)
+  const balancesPerNetwork = useBalancesFiatTotalPerNetwork(balances)
+
+  const compatibleChains = useMemo<NetworkWithExplorer[]>(
+    () =>
+      chains
+        .filter(
+          (chain) =>
+            !!chain.subscanUrl &&
+            !!chain.name &&
+            // account is undefined for contacts
+            (account
+              ? isAccountCompatibleWithChain(chain, account.type ?? "sr25519", account.genesisHash)
+              : isAddressCompatibleWithChain(chain, address))
+        )
+        .map(
+          (chain): NetworkWithExplorer => ({
+            id: chain.id,
+            type: "substrate",
+            name: chain.name!,
+            explorerUrl: chain.subscanUrl!,
+          })
+        ),
+    [account, address, chains]
+  )
+
+  const compatibleEvmNetworks = useMemo<NetworkWithExplorer[]>(
+    () =>
+      isEthereumAddress(address)
+        ? evmNetworks
+            .filter((network) => !!network.name && !!network.explorerUrl)
+            .map(
+              (chain): NetworkWithExplorer => ({
+                id: chain.id,
+                type: "substrate",
+                name: chain.name!,
+                explorerUrl: chain.explorerUrl!,
+              })
+            )
+        : [],
+    [address, evmNetworks]
+  )
+
+  const sortedNetworks = useMemo(
```

Comment:
Maybe not worth worrying about for this time, but perhaps in future we could extend the sorting method to include crypto balances with no fiat value - so, first up would be chains with fiat value, then next would be chains with some token balances without any known fiat value, then finally chains sorted in alphabetical order.